### DISCUSSION TITLE
get the thermal conductivity of the material with pyaedt scripts

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4660] [Q&A] [answered]
Hi all, I am facing a problem while post-processing using pyaedt script. The issue is that I need the thermal conductivity of the material in my report, so I want to get the thermal conductivity of the material with pyaedt scripts.
  I tried the code but it reports an error.
```
material_name = "plastic"
material_thermal_conductivity = ipk.materials[material_name].thermal_conductivity
```
  I would like to know which pyaedt script I need to use to get the thermal conductivity of a material!
  Looking forward to your reply. Thanks.

**Comment by Samuelopez-ansys** [comment, marked as answer]
@aknsj 

I assume, the material "plastic" is used in your design, because if it is not, then you can not get it in the way you are trying:

<img width="233" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/3abd0ed2-86f8-4aaf-9259-340c07751b0b">

If the material is defined like in the previous picture.
<img width="717" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/3693b050-af3f-45cc-9433-d2fd4b1bf547">

<img width="517" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/ee91b227-d58e-4538-a589-2ea4ff431580">


If the material is not defined in the design, you need to first load it, 

app.materials.add_material("polyamide")

If the material name you pass exists in the AEDT library, then the material definition will be the same, if you pass a new name, you will create a dummy material, that you can customize later. This is how AEDT works:
<img width="238" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/f4736ba4-7cd1-4274-a192-13427c059f0f">


Please let me know if this solves your questions.

**Reply by aknsj** [reply]
Thank you for your detailed reply. From your reply I not only learned the difference between whether the name I passed in exists in the material library or not while adding a material with pyaedt, but I also learned how to get the thermal conductivity of a material that has been created.
  I successfully got the thermal conductivity value of the material using the following code for me to use in my report.
```
material_name_1 = "plastic"
material_name_2 = "new_material"
ipk.materials.add_material(material_name_1)
ipk.materials.add_material(material_name_2).thermal_conductivity = 0.1

material_thermal_conductivity = ipk.materials[material_name_1].thermal_conductivity.evaluated_value
```
  Thank you very much for your reply which solved my problem!
